support/oidc-discovery-provider/main.go

```diff
@@ -140,6 +167,25 @@ func newSource(log logrus.FieldLogger, config *Config) (JWKSSource, error) {
 	}
 }
 
+func newListenerWithServingCert(ctx context.Context, log logrus.FieldLogger, config *Config) (net.Listener, error) {
+	certManager, err := NewDiskCertManager(config.ServingCertFile, log)
+	if err != nil {
+		return nil, err
+	}
+	go func() {
+		certManager.WatchFileChanges(ctx)
+	}()
+
+	tlsConfig := certManager.TLSConfig()
+
+	tcpListener, err := net.ListenTCP("tcp", &net.TCPAddr{Port: 8080})
```

Comment:
is this hardcoded port ok?

Reply:
It should be 443

Reply:
The acme code for sure used 443. Now that we're not bound to ACME, this could probably be configurable so you don't need `CAP_NET_BIND_SERVICE` capabilities when testing it out...

Reply:
You're right, we can provide more flexibility now, I will introduce a new addr configuration in this section.